--- sparse-0.6.1/Makefile.orig	2019-10-14 01:27:22.000000000 +0200
+++ sparse-0.6.1/Makefile	2020-04-03 01:52:37.902999893 +0200
@@ -161,7 +161,7 @@
 endif
 ifneq ($(filter ${arch},i386 i486 i586 i686 x86_64 amd64),)
 LLVM_VERSION:=$(shell $(LLVM_CONFIG) --version)
-ifeq ($(shell expr "$(LLVM_VERSION)" : '[3-9]\.'),2)
+ifeq ($(shell expr $(shell expr "$(LLVM_VERSION)" : '\([0-9]\+\)\.') '>' 2),1)
 LLVM_PROGS := sparse-llvm
 $(LLVM_PROGS): LD := g++
 LLVM_LDFLAGS := $(shell $(LLVM_CONFIG) --ldflags)
